From fcc3187b11b9d0e979a702503104cfaac8b5eab4 Mon Sep 17 00:00:00 2001
From: Johannes Schindelin <johannes.schindelin@gmx.de>
Date: Fri, 1 Mar 2019 13:00:32 +0100
Subject: [PATCH 53/N] fixup! Add functionality for changing OS name via
 MSYSTEM environment variables.

MSYS2 wants to add one extra byte to the `utsname` struct, to make sure
that `MINGW64` fits where `CYGWIN` would have been printed before.

In Cygwin v3.0.0, the `utsname` struct was changed, to allow for much
longer names. We also extended that when rebasing, but we failed to
re-extend the `old_utsname` which is there for previously-compiled
executables that do not know about the new `struct utsname`.

Let's change also `old_utsname`.

This fixes https://github.com/git-for-windows/git/issues/2081

Signed-off-by: Johannes Schindelin <johannes.schindelin@gmx.de>
---
 winsup/cygwin/uname.cc | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/winsup/cygwin/uname.cc b/winsup/cygwin/uname.cc
index 2df6e9a..a55f719 100644
--- a/winsup/cygwin/uname.cc
+++ b/winsup/cygwin/uname.cc
@@ -76,7 +76,7 @@ uname_x (struct utsname *name)
 /* Old entrypoint for applications up to API 334 */
 struct old_utsname
 {
-  char sysname[20];
+  char sysname[21];
   char nodename[20];
   char release[20];
   char version[20];
-- 
2.9.0

